File: CLI To-Do App/Golang/tasks/validate.go
package tasks

// Validate task ID method
func (t *Tasks) ValidateTaskID(id int) error {
	if id < 1 || id > len(*t) {
		return ErrorInvalidTaskID
	}

	if !(*t)[id-1].Visible {
		return ErrorTaskNotFound
	}

	return nil
}

File: CLI To-Do App/Rust/src/tasks/validate.rs
use super::types::{TaskError, Tasks};

// Tasks implementation
impl Tasks {
    // Validate Task ID method
    pub fn validate_task_id(&self, id: usize) -> Result<(), TaskError> {
        if id < 1 || id > self.len() {
            return Err(TaskError::InvalidTaskId);
        }

        if !self[id - 1].visible {
            return Err(TaskError::TaskNotFound);
        }

        Ok(())
    }
}
